server/src/v2/middlewares/multer.js

```diff
@@ -0,0 +1,38 @@
+import multer from 'multer';
+import fileType from 'file-type';
+
+const acceptedExtensions = ['jpg', 'png'];
+
+const storage = multer.memoryStorage();
+
+const imageFormatValidator = (req, res, next) => {
+  if (req.file) {
+    if (req.file.buffer) {
+      // For MemoryStorage, validate the format using `req.file.buffer`
+      const mime = fileType(req.file.buffer);
+
+      // if can't be determined or format not accepted
+      if (!mime || !acceptedExtensions.includes(mime.ext)) { return next(new Error(`The uploaded file is not in  ${acceptedExtensions.join(',')} format!`)); }
```

Comment:
Hey @JackieBinya you did not fix this return statements to use the response for the errors. As it is it will still crash the app